src/io/memory.cr

```diff
@@ -225,7 +225,9 @@ class IO::Memory
     if pos == @bytesize
       ""
     else
-      String.new(@buffer + @pos, @bytesize - @pos)
+      String.new(@buffer + @pos, @bytesize - @pos).tap {
```

Comment:
In my opinion, the stdlib should stick to one rule, and that rule should be easy to enforce.